File: src/utils/hash.ts
import crypto from 'crypto';

/**
 * Generate a stable SHA-256 hash for a given input.
 */
export function hashString(input: string): string {
    return crypto
        .createHash('sha256')
        .update(input)
        .digest('hex');
}

/**
 * Hash multiple strings as a single deterministic key.
 */
export function hashStrings(inputs: string[]): string {
    return hashString(inputs.join('||'));
}

File: rust/rlm/src/utils/hash.rs
#![allow(dead_code)]

use sha2::{Digest, Sha256};

pub fn hash_string(input: &str) -> String {
    let mut hasher = Sha256::new();
    hasher.update(input.as_bytes());
    let result = hasher.finalize();
    hex::encode(result)
}

pub fn hash_strings(inputs: &[String]) -> String {
    hash_string(&inputs.join("||"))
}
